compiler/rustc_passes/src/dead.rs

```diff
@@ -722,6 +784,41 @@ impl<'tcx> Visitor<'tcx> for MarkSymbolVisitor<'tcx> {
 
         self.in_pat = in_pat;
     }
+
+    fn visit_poly_trait_ref(&mut self, t: &'tcx hir::PolyTraitRef<'tcx>) {
+        // mark the assoc type/const appears in poly-trait-ref live
+        if let Some(pathsegment) = t.trait_ref.path.segments.last()
```

Comment:
this comes from #127505, and this PR should be merged after that

Reply:
Feel free to take it, and the tests etc. from that PR. You've got a better handle on this anyway. :)